x/vm/keeper/state_transition.go

```diff
@@ -223,25 +211,25 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, tx *ethtypes.Transaction) (*t
 	}
 
 	ethLogs := types.LogsToEthereum(res.Logs)
-	_, bloomReceipt := k.initializeBloomFromLogs(ctx, ethLogs)
+	// Compute block bloom filter
+	if len(ethLogs) > 0 {
+		k.SetTxBloom(tmpCtx, new(big.Int).SetBytes(logsBloom(ethLogs)))
+	}
 
 	var contractAddr common.Address
 	if msg.To == nil {
 		contractAddr = crypto.CreateAddress(msg.From, msg.Nonce)
 	}
 
 	receipt := &ethtypes.Receipt{
-		Type:              tx.Type(),
-		PostState:         nil,
-		CumulativeGasUsed: calculateCumulativeGasFromEthResponse(ctx.GasMeter(), res),
-		Bloom:             bloomReceipt,
-		Logs:              ethLogs,
-		TxHash:            txConfig.TxHash,
-		ContractAddress:   contractAddr,
-		GasUsed:           res.GasUsed,
-		BlockHash:         common.BytesToHash(ctx.HeaderHash()),
-		BlockNumber:       big.NewInt(ctx.BlockHeight()),
-		TransactionIndex:  txConfig.TxIndex,
+		Type:            tx.Type(),
+		PostState:       nil,
+		Logs:            ethLogs,
+		TxHash:          txConfig.TxHash,
+		ContractAddress: contractAddr,
+		GasUsed:         res.GasUsed,
+		BlockHash:       common.BytesToHash(ctx.HeaderHash()),
+		BlockNumber:     big.NewInt(ctx.BlockHeight()),
```

Comment:
is CumulativeGasUsed set elsewhere? does this break anyone?

Reply:
Good call, I think the main use case for all of these fields in the receipt is to send them to the post tx processing hooks. I've added back in all of the fields except for the bloom data which isn't updated until the end of the block anymore so intermediate changes to the block bloom wouldn't change things and the correct value will be calculated based on the included logs.